Assets/Scripts/Bullet.cs

```diff
@@ -30,4 +31,21 @@ void Explode()
     {
         Destroy(gameObject);
     }
+
+    /**
+     * Damage every killable thing that it this
+     */
+    void OnCollisionEnter2D(Collision2D other) {
+        MonoBehaviour[] behaviours =
+            other.gameObject.GetComponents<MonoBehaviour>();
+        
+        foreach(MonoBehaviour behaviour in behaviours) {
+            if (behaviour is IAlive) {
+                IAlive killable = (IAlive) behaviour;
+                killable.TakeDamage(damage);
+            }
+        }
+
+        Explode();
+    }

```

Comment:
You could just do this:

```cs
    void OnCollisionEnter2D(Collision2D other) {
        IAlive killable = other.GetComponent<IAlive>();
        if (killable) {
            killable.TakeDamage(damage);
        }

        Explode();
    }
```
But using interfaces is really interesting, instead of using tags for example. Thank you for that!

Reply:
I read that GetComponent should not be used for interfaces. Behaviour with interface is inconsistent (GetComponents and GetComponent does not react the same when using interface) so I went for the generic approach :/